### ISSUE TITLE
After a single edit of a JS file, red squigglies stay permanently outdated.

### ISSUE THREAD

**Issue by trusktr** [microsoft/TypeScript#37955]
**TypeScript Version:**  3.8.3 (VS Code 1.43.2 Commit 0ba0ca52)

**Search Terms:**

typescript red squigglies do not update after editing

**Code**

tsconfig.json:

```jsonc
{
  "compilerOptions": {
    "noEmit": true,

    "checkJs": true,
    "allowJs": true,

    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true
  },
  "include": ["./wasmaudioworklet/**/*"]
}
```

**Expected behavior:**

When I open VS Code on the project in question, the initial type check goes well, and red squigglies appear where expected in plain JS files.

I expect that any time I edit the JS files, the red squigglies update.

**Actual behavior:**

As soon as I make a single edit, red squigglies all appear to stay outdated.

**Playground Link:**

N/A

**Related Issues:**

https://github.com/microsoft/TypeScript/issues/6836

** Reproduction: **

```
git clone git@github.com:trusktr/javascriptmusic.git
cd javascriptmusic
git checkout typescript-issue-37955
```

Then open VS Code (that version I listed uses the builtin TypeScript at version 3.8.3). Now open `wasmaudioworklet/app.js`, then see any errors there or in any of its imports, try to fix them, and the red squigglies should fail to update.

That's what I'm doing at least, and I am able to reproduce every time without failing.

Please note, I did not run `npm install`, there is no local `node_modules` folder while this problem happens.

Here's more version info:

```
Version: 1.43.2
Commit: 0ba0ca52957102ca3527cf479571617f0de6ed50
Date: 2020-03-25T01:43:18.461Z
Electron: 7.1.14
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 5.4.28-1-MANJARO
```

**Comment by trusktr** [comment]
I added a reproduction to the original post.

**Comment by sheetalkamat** [comment]
With 3.8.3 i am seeing following exception which is fixed by #37545 and i dont see the issue repro in master any more.
```
Err 154   [11:28:7.978] Exception on executing command semanticDiag:

    Debug Failure. False expression: Expected position undefined to be a number.

    Error: Debug Failure. False expression: Expected position undefined to be a number.
        at failIfInvalidPosition (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:138893:22)
        at ScriptInfo.positionToLineOffset (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:138870:17)
        at formatDiag (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:143839:35)
        at c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:144798:100
        at Array.map (<anonymous>)
        at IOSession.Session.sendDiagnosticsEvent (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:144798:71)
        at IOSession.Session.semanticCheck (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:144788:22)
        at c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:144843:31
        at MultistepOperation.executeAction (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:143950:25)
        at c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:143931:100
        at IOSession.Session.executeWithRequestId (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:145994:28)
        at Object.executeWithRequestId (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:144583:87)
        at Immediate._onImmediate (c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:143931:41)
        at processImmediate (internal/timers.js:439:21)
```

**Comment by sheetalkamat** [comment]
Duplicate of #37084
